src/BenchmarkDotNet/Toolchains/MonoWasm/WasmAppBuilder.cs

```diff
@@ -0,0 +1,201 @@
+    }
+
+    public bool BuildApp (string programName, string projectRoot)
+    {
+        string appDir = Path.Combine(projectRoot, $"bin/{TargetFrameworkMoniker}/browser-wasm/publish");
+
+        string outputDir = Path.Combine(appDir, "output");
+
+        List<string> assemblySearchPaths = new List<string>
+                                  {
+                                    appDir
+                                  };
+
+        string mainAssemblyPath = Path.Combine(appDir, $"{programName}.dll");
+
+
+        if (!File.Exists(mainAssemblyPath))
+            throw new ArgumentException($"File MainAssembly='{mainAssemblyPath}' doesn't exist.");
+        if (!File.Exists(WasmSettings.WasmMainJS))
+            throw new ArgumentException($"File MainJS='{WasmSettings.WasmMainJS}' doesn't exist.");
+
+        var paths = new List<string>();
+        _assemblies = new Dictionary<string, Assembly>();
+
+        // Collect and load assemblies used by the app
+        foreach (string dir in assemblySearchPaths)
+        {
+            if (!Directory.Exists(dir))
+                throw new ArgumentException($"Directory '{dir}' doesn't exist or not a directory.");
+            paths.Add(dir);
+        }
+        _resolver = new AssemblyResolver(paths);
+        var mlc = new MetadataLoadContext(_resolver, "System.Private.CoreLib");
+
+        Assembly mainAssembly = mlc.LoadFromAssemblyPath (mainAssemblyPath);
+        Add(mlc, mainAssembly);
+
+        // Create app
+        Directory.CreateDirectory(outputDir);
+        Directory.CreateDirectory(Path.Combine(outputDir, "managed"));
+        foreach (var assembly in _assemblies.Values)
+            File.Copy(assembly.Location, Path.Combine(outputDir, "managed", Path.GetFileName(assembly.Location)), true);
+        foreach (var f in new string[] { "dotnet.wasm", "dotnet.js" })
+            File.Copy(Path.Combine(appDir, f), Path.Combine(outputDir, f), true);
+        File.Copy(WasmSettings.WasmMainJS, Path.Combine(outputDir, "runtime.js"),  true);
+
+        string supportFilesDir = Path.Combine(outputDir, "supportFiles");
+        Directory.CreateDirectory(supportFilesDir);
+
+        var filesToMap = copySystemPrivateCoreLib(appDir, supportFilesDir);
+
+        using (var sw = File.CreateText(Path.Combine(outputDir, "mono-config.js")))
+        {
+            sw.WriteLine("config = {");
+            sw.WriteLine("\tvfs_prefix: \"managed\",");
+            sw.WriteLine("\tdeploy_prefix: \"managed\",");
+            sw.WriteLine("\tenable_debugging: 0,");
+            sw.WriteLine("\tassembly_list: [");
+            foreach (var assembly in _assemblies.Values)
+            {
+                sw.Write("\t\t\"" + Path.GetFileName(assembly.Location) + "\"");
+                sw.WriteLine(",");
+            }
+            sw.WriteLine ("\t],");
+            sw.WriteLine("\tfiles_to_map: [");
+            foreach (KeyValuePair<string, List<string>> keyValuePair in filesToMap)
```

Comment:
I'm pretty sure we don't need to have any entries in `files_to_map` so this can be removed.